support/refc/stringOps.c

```diff
@@ -77,8 +77,8 @@ Value *strSubstr(Value *start, Value *len, Value *s) {
   int l = extractInt(len);
 
   int tailLen = strlen(input);
-  if (tailLen < l) {
-    l = tailLen;
+  if (tailLen < offset + l) {
+    l = tailLen - offset;
```

Comment:
Could you add a test for this fix?

Reply:
Added it.

It is not visually apparent because the NUL character is copied and the result is visually correct.
Therefore, it is difficult to detect in testing.

I never noticed this bug until I saw valgrind's report.